contracts/mocks/CloneRewarderTime.sol

```diff
@@ -0,0 +1,153 @@
+    using BoringERC20 for IERC20;
+
+    IERC20 public rewardToken;
+
+    /// @notice Info of each MCV2 user.
+    /// `amount` LP token amount the user has provided.
+    /// `rewardDebt` The amount of SUSHI entitled to the user.
+    struct UserInfo {
+        uint256 amount;
+        uint256 rewardDebt;
+    }
+
+    /// @notice Info of each MCV2 pool.
+    /// `allocPoint` The amount of allocation points assigned to the pool.
+    /// Also known as the amount of SUSHI to distribute per block.
+    struct PoolInfo {
+        uint128 accSushiPerShare;
+        uint64 lastRewardTime;
+    }
+
+    /// @notice Info of each pool.
+    mapping (uint256 => PoolInfo) public poolInfo;
+
+
+    /// @notice Info of each user that stakes LP tokens.
+    mapping (uint256 => mapping (address => UserInfo)) public userInfo;
+
+    uint256 public rewardPerSecond;
+    IERC20 public masterLpToken;
+    uint256 private constant ACC_TOKEN_PRECISION = 1e12;
+
+    address public immutable MASTERCHEF_V2;
+
+    event LogOnReward(address indexed user, uint256 indexed pid, uint256 amount, address indexed to);
+    event LogPoolAddition(uint256 indexed pid, uint256 allocPoint);
+    event LogSetPool(uint256 indexed pid, uint256 allocPoint);
+    event LogUpdatePool(uint256 indexed pid, uint64 lastRewardTime, uint256 lpSupply, uint256 accSushiPerShare);
+    event LogRewardPerSecond(uint256 rewardPerSecond);
+    event LogInit(IERC20 indexed rewardToken, address owner, uint256 rewardPerSecond, IERC20 indexed masterLpToken);
+
+    constructor (address _MASTERCHEF_V2) public {
+        MASTERCHEF_V2 = _MASTERCHEF_V2;
+    }
+
+    /// @notice Serves as the constructor for clones, as clones can't have a regular constructor
+    /// @dev `data` is abi encoded in the format: (IERC20 collateral, IERC20 asset, IOracle oracle, bytes oracleData)
+    function init(bytes calldata data) public payable {
+        require(rewardToken == IERC20(0), "Rewarder: already initialized");
+        (rewardToken, owner, rewardPerSecond, masterLpToken) = abi.decode(data, (IERC20, address, uint256, IERC20));
+        require(rewardToken != IERC20(0), "Rewarder: bad token");
+        emit LogInit(rewardToken, owner, rewardPerSecond, masterLpToken);
+    }
+
+    function onSushiReward (uint256 pid, address _user, address to, uint256, uint256 lpToken) onlyMCV2 override external {
+        require(IMasterChefV2(MASTERCHEF_V2).lpToken(pid) == masterLpToken);
+
+        PoolInfo memory pool = updatePool(pid);
+        UserInfo storage user = userInfo[pid][_user];
+        uint256 pending;
+        if (user.amount > 0) {
+            pending =
+                (user.amount.mul(pool.accSushiPerShare) / ACC_TOKEN_PRECISION).sub(
+                    user.rewardDebt
+                );
+            rewardToken.safeTransfer(to, pending);
+        }
+        user.amount = lpToken;
+        user.rewardDebt = lpToken.mul(pool.accSushiPerShare) / ACC_TOKEN_PRECISION;
```

Comment:
Although it's not a problem here due to how MCv2 works, the safe transfer call should come after we've updated user's reward debt and amount to prevent reentrancy attacks.

Reply:
You are right, this is kept close to the original implementation